### PR TITLE
feat(uibuilder): nested collections support and codegen dep update

### PR DESCRIPTION
**Pull request by SwaySway**
#### Description of changes

- updates lastest version of codegen `v2.3.2`
- use `AmplifyClientFactory` to use same aws creds for AmplifyUIBuilder and AmplifyBackend
- fetch BackendModels to create GenericDataSchema to be used with component generation
- updated unit tests
- allow endpoint and region to be overridden in AmplifyBackend `generateBackendModels` call
  - use env variables `AMPLIFY_BACKEND_ENDPOINT` &  `AMPLIFY_BACKEND_REGION`

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)
- [x] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-util-uibuilder/src/commands/cloneComponentsFromEnv.ts [hunk 1]
```diff
@@ -24,35 +28,29 @@ export async function run(context: $TSContext) {
...
-          properties: component.properties,
-          sourceId: component.sourceId,
-          variants: component.variants,
-        },
-      })
-      .promise();
-  });
+  const promises = components.map(component => studioClient.createComponent(
+    {
+      bindingProperties: component.bindingProperties,
```
Comment (cwoolum) [inline packages/amplify-util-uibuilder/src/commands/cloneComponentsFromEnv.ts:38]: Why not just spread `component`?

File: packages/amplify-util-uibuilder/src/commands/cloneComponentsFromEnv.ts [hunk 2]
```diff
@@ -24,35 +28,29 @@ export async function run(context: $TSContext) {
...
-          name: component.name,
-          overrides: component.overrides,
-          properties: component.properties,
-          sourceId: component.sourceId,
-          variants: component.variants,
-        },
-      })
-      .promise();
-  });
+  const promises = components.map(component => studioClient.createComponent(
```
Comment (cwoolum) [inline packages/amplify-util-uibuilder/src/commands/cloneComponentsFromEnv.ts:36]: We need to make sure we throttle this so we don't get 429'd

File: packages/amplify-util-uibuilder/src/commands/generateComponents.ts [hunk 3]
```diff
@@ -50,7 +53,11 @@ export const run = async (context: $TSContext): Promise<void> => {
 
     const invalidComponentNames = componentSchemas.entities.filter(component => !component.schemaVersion).map(component => component.name);
     if (invalidComponentNames.length) {
-      printer.warn(`The components ${invalidComponentNames.join(', ')} were synced with an older version of Amplify Studio. Please re-sync your components with Figma to get latest features and changes.`);
+      printer.warn(
+        `The components ${invalidComponentNames.join(
```
Comment (cwoolum) [inline packages/amplify-util-uibuilder/src/commands/generateComponents.ts:57]: Is there any sort of localization in the CLI or is it only English?

**Review by cwoolum — CHANGES_REQUESTED** [review]
Left a few comments

File: packages/amplify-util-uibuilder/src/commands/generateComponents.ts (hunk 3 above)
Comment (SwaySway) [reply packages/amplify-util-uibuilder/src/commands/generateComponents.ts:57]: Not currently

File: packages/amplify-util-uibuilder/src/commands/cloneComponentsFromEnv.ts (hunk 2 above)
Comment (SwaySway) [reply packages/amplify-util-uibuilder/src/commands/cloneComponentsFromEnv.ts:36]: We can await each creation and run that sequentially as opposed to promise.all or do you have preferred rate to limit batch requests?

**Review by cwoolum — APPROVED** [review]
LGTM

File: packages/amplify-cli/src/amplify-service-helper.ts [hunk 4]
```diff
@@ -1,9 +1,14 @@
...
+import {
+  $TSObject, $TSContext, pathManager, stateManager,
+} from 'amplify-cli-core';
 import { isDataStoreEnabled } from 'graphql-transformer-core';
-import { normalizeInputParams } from './input-params-manager';
 import * as path from 'path';
 import _ from 'lodash';
+import { normalizeInputParams } from './input-params-manager';
 
+/**
```
Comment (lazpavel) [inline packages/amplify-cli/src/amplify-service-helper.ts:9]: can we add a description or remove the empty comments?

**Comment by SwaySway** [comment]
cloud-e2e run: https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli?branch=run-e2e/josurui/feat/nested-collections

File: packages/amplify-util-uibuilder/src/index.ts [hunk 5]
```diff
@@ -1,16 +1,26 @@
 import { $TSContext } from 'amplify-cli-core';
 import path from 'path';
 
-export async function executeAmplifyCommand(context: $TSContext, cmd: string) {
+// TODO: remove dynamic require
```
Comment (lazpavel) [inline packages/amplify-util-uibuilder/src/index.ts:4]: is this intended to be done in this PR?

**Review by lazpavel — APPROVED** [review]
LGTM, 2 small nits/questions

File: packages/amplify-util-uibuilder/src/index.ts (hunk 5 above)
Comment (SwaySway) [reply packages/amplify-util-uibuilder/src/index.ts:4]: nope we want to scope that in different pr

**Pull request merged**